From 1869efdef3bb1889153837ca8d22aaa2fe4054e6 Mon Sep 17 00:00:00 2001
From: Robert Dabrowski <robertx.dabrowski@intel.com>
Date: Thu, 11 Jun 2020 02:04:29 +0200
Subject: [PATCH] PONRTSYS-7543: mib_counter show crash in system with VUNI
 driver  -replace direct get mac_ops by function with range check

---
 drivers/net/datapath/mib_counters/dp_mib_counters.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/datapath/mib_counters/dp_mib_counters.c b/drivers/net/datapath/mib_counters/dp_mib_counters.c
index 995d4a88ee94..c842ca5a1424 100644
--- a/drivers/net/datapath/mib_counters/dp_mib_counters.c
+++ b/drivers/net/datapath/mib_counters/dp_mib_counters.c
@@ -216,7 +216,7 @@ static void update_dev_counters(struct mib_counters *cnt, u8 initial_update)
 	}
 
 	/* mac_ops is optional, do not fail if unavailable */
-	mac_ops = dp_port_prop[cnt->inst].mac_ops[cnt->port_id];
+	mac_ops = gsw_get_mac_ops(cnt->inst, cnt->port_id);
 	if (mac_ops) {
 		rmon = devm_kzalloc(&g_dp_dev->dev, sizeof(struct mac_rmon),
 				    GFP_ATOMIC);
